kratos/sources/model_part_io.cpp

```diff
@@ -138,10 +131,14 @@ std::size_t ModelPartIO::ReadNodesNumber()
 
 void ModelPartIO::WriteNodes(NodesContainerType const& rThisNodes)
 {
+    // Printing or not with scientific precision
+    if (mOptions.Is(IO::SCIENTIFIC_PRECISION)) {
+        (*mpStream) << std::setprecision(10) << std::scientific;
+    }
     (*mpStream) << "Begin Nodes" << std::endl;
-    for(NodesContainerType::const_iterator it_node = rThisNodes.begin() ; it_node != rThisNodes.end() ; it_node++)
-        (*mpStream) << "\t" << it_node->Id() << "\t" << it_node->X()  << "\t" << it_node->Y() << "\t" << it_node->Z() << std::endl;
-    (*mpStream) << "End Nodes" << std::endl << std::endl;
```

Comment:
I would not change it to the ascii return

Reply:
Okidoki

Reply:
(a priori it is better https://www.educative.io/edpresso/what-is-the-difference-between-endl-and-n-in-cpp)

Reply:
Done

Reply:
I strongly recommend using `\n`
See #7922

Reply:
I think that @pooyan-dadvand referred to the last two outside the loop.

Reply:
okay, done